sim/core/spell_mod.go

```diff
@@ -66,9 +73,16 @@ func buildMod(unit *Unit, config SpellModConfig) *SpellMod {
 
 		applyFn = config.ApplyCustom
 		removeFn = config.RemoveCustom
+
+		if config.OnReset != nil {
+			resetFn = config.OnReset
+		} else {
+			resetFn = functions.OnReset
+		}
```

Comment:
This can be simplified to just
```
resetFn = config.ResetCustom
```
since `functions.OnReset` will just be `nil` in this case regardless.